frontend/src/app/user/service/user-role.service.ts

```diff
@@ -0,0 +1,24 @@
+import {Injectable} from '@angular/core';
+import {ActivatedRouteSnapshot, RouterStateSnapshot} from '@angular/router';
+import {UserByEmailResolverService} from './user-by-email-resolver.service';
+import {Observable} from 'rxjs/Observable';
+
+@Injectable()
+export class UserRoleService {
+
+    constructor(private userByEmailResolverService: UserByEmailResolverService) { }
+
+    /**
+     * determine if the current user has any role in list
+     */
+    hasRoles(list: [string]): Observable<boolean> {
+        return this.userByEmailResolverService.resolve().flatMap((v) => {
+            const roles = v._embedded.users[0].authorities.map(r => r.authority);
+            let result = false;
+            list.forEach(r => {
+                result = result || roles.includes(r);
```

Comment:
This is an ok use of using single letter variable name. `roles.includes(r)` implies that `r` is a `role` without me having to dig into the api. Whereas previously mapping an authority was ambiguous (without digging into the API)